b9lab-content/3-main-concepts/10-multistore-keepers.md

```diff
@@ -1,5 +1,115 @@
 # Multistore & Keepers
 
+A Cosmos SDK application on a purpose-built blockchain usually consists of several modules attending to separate concerns. Each module has a state that is a subset of the entire application state. 
+
+Cosmos SDK adopts an object-capabilities-based approach (only reveal what is necessary to get the work done) to help developers better protect their application from unwanted inter-module interactions, and keepers are at the core of this approach.
+
+A keeper is a Cosmos SDK abstraction that manages access to the subset of the state defined by a module. All access to the module’s data must go through the module’s keeper. 
+
+A keeper can be thought of quite literally as the gatekeeper of a module's store(s). Each store (typically an IAVL Store) defined within a module comes with a storeKey, which grants unlimited access to it. The module's keeper holds this storeKey (which should otherwise remain unexposed), and defines methods for reading and writing to the store(s).
+
+When a module needs to interact with the state defined in another module, it does so by interacting with methods of the other module’s keeper. Developers control the interactions their module can have with other modules by defining methods and controlling access. 
+
+## Format
+
+Keepers are generally defined in a `/keeper/Keeper.go` file locating in the module’s folder.
+
+By convention, the type keeper of a module is named simply `Keeper.go` and usually follows the following structure:
+
+```
+type Keeper struct {
+    // Expected external keepers, if any
+
+    // Store key(s)
+
+    // codec
+}
+```
+
+### Parameters:
+
+* An `expected` keeper is a keeper external to a module that is required by the internal keeper of said module. External keepers are listed in the internal keeper's type definition as interfaces. These interfaces are themselves defined in an `expected_keepers.go` file in the root of the module's folder. In this context, interfaces are used to reduce the number of dependencies, as well as to facilitate the maintenance of the module itself.
+* `storeKeys` grant access to the store(s) of the multistore managed by the module. They should always remain unexposed to external modules.
+* `cdc` is the codec used to marshall and unmarshall structs to/from []byte. The cdc can be any of codec.BinaryCodec, codec.JSONCodec or codec.Codec based on your requirements. It can be either a proto or amino codec as long as they implement these interfaces.
```

Comment:
I think we should mention that this is only `codec.Codec` as it implements the other interfaces.